### PR TITLE
build: run CI unit tests using bazel

### PR DESCRIPTION
**Pull request by josephperrott**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "bazel:buildifier": "find . -type f \\( -name \"*.bzl\" -or -name WORKSPACE -or -name BUILD -or -name BUILD.bazel \\) ! -path \"*/node_modules/*\" | xargs buildifier -v --warnings=attr-cfg,attr-license,attr-non-empty,attr-output-default,attr-single-file,constant-glob,ctx-args,depset-iteration,depset-union,dict-concatenation,duplicated-name,filetype,git-repository,http-archive,integer-division,load,load-on-top,native-build,native-package,output-group,package-name,package-on-top,redefined-variable,repository-name,same-origin-load,string-iteration,unused-variable",
     "bazel:format-lint": "yarn -s bazel:buildifier --lint=warn --mode=check",
     "dev-app": "gulp serve:devapp",
-    "test": "bazel test //src/... --test_tag_filters=-e2e",
+    "test": "bazel test //src/... --test_tag_filters=-e2e,-browser:firefox-local --build_tag_filters=-browser:firefox-local --build_tests_only",
```
Comment (jelbourn) [inline package.json]: Add another command for `test-firefox`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -144,7 +144,6 @@ jobs:
       - *yarn_install
 
       - run: yarn bazel build src/... --build_tag_filters=-docs-package
```
Comment (devversion) [inline .circleci/config.yml]: should we rename the job above to `bazel_build` instead?

Comment (josephperrott) [reply .circleci/config.yml]: Sure!

**Comment by josephperrott** [comment]
Caretaker Note: When this goes in we will need to update our required statuses for github to include `bazel_build` instead of `bazel_build_test`

**Pull request merged**